src/test/fuzz/addrman.cpp

```diff
@@ -238,6 +241,9 @@ FUZZ_TARGET_INIT(addrman, initialize_addrman)
         ds.SetVersion(ser_version);
         try {
             ds >> *addr_man_ptr;
+            if (0 != WITH_LOCK(addr_man_ptr->cs, return addr_man_ptr->ForceCheckAddrman())) {
+                throw std::ios_base::failure{"Reset addrman after consistency check failed"};
+            }
```

Comment:
What about moving this unconditional check inside, at the end of `CAddrMan::Unserialize()`? We never want to continue if this check fails after unserialize and it is not too expensive to check once per unserialize.

Right now we have the conditional `Check()` at the end of `Unserialize()`.

Reply:
This is what #22734 is doing. Happy to close this fuzz-only pull and rebase the other.

Reply:
Up to you (and other reviewers). I "voted" for the other PR :)